src/ast/async_ids.h

```diff
@@ -12,6 +12,7 @@ namespace ast {
   DO(bpf_print)                                                                \
   DO(non_map_print)                                                            \
   DO(printf)                                                                   \
+  DO(read_map_value)                                                           \
```

Comment:
I'm not familiar with this file, but something feels wrong here. Async events are things that we pass over the ring buffer for processing in userspace.

`read_map_value`, `tuple` and `str` aren't async events as they're all handled synchronously in BPF-space. Possibly also not `bpf_print`.

Reply:
AsyncIDs class provides functionality to keep a running counter. This allows us to use this to assign memory slots for tuple/str/read_map_value scratch buffers, so that we make sure one usage of the buffer doesn't overwrite the buffer. See my comment here: https://github.com/bpftrace/bpftrace/pull/3456#issuecomment-2387264451

I am happy to rename this class to `Counter` or `IdGenerator` or something like that

Reply:
Ok yeah, if this doesn't represent anything "async", then the name is very misleading!

It looks like the class originally was just for async events IDs, which have to stay in sync with the corresponding code in resource_analyser.cpp and are used in bpftrace.cpp. Some comments explaining the purpose of this class would be nice.

If the class is no longer just for async events, then a name change would make sense.

Reply:
@ajor if you don't mind, I will do the rename in a separate PR, I prefer smaller PRs and this one already has some non-trivial complexities :)